addons/mail/static/tests/thread/thread.test.js

```diff
@@ -110,20 +119,34 @@ test("do not show message subject when subject is the same as the thread name",
     });
 });
 
-test("auto-scroll to bottom of thread on load", async () => {
+test("auto-scroll to last read message on thread load", async () => {
     const pyEnv = await startServer();
     const channelId = pyEnv["discuss.channel"].create({ name: "general" });
-    for (let i = 1; i <= 25; i++) {
-        pyEnv["mail.message"].create({
-            body: "not empty",
-            model: "discuss.channel",
-            res_id: channelId,
-        });
+    const messageIds = [];
+    for (let i = 0; i <= 200; i++) {
+        messageIds.push(
+            pyEnv["mail.message"].create({
+                body: `message ${i}`,
+                model: "discuss.channel",
+                res_id: channelId,
+            })
+        );
     }
+    const [selfMemberId] = pyEnv["discuss.channel.member"].search([
+        ["partner_id", "=", serverState.partnerId],
+        ["channel_id", "=", channelId],
+    ]);
+    pyEnv["discuss.channel.member"].write([selfMemberId], {
+        new_message_separator: messageIds[100],
+    });
     await start();
     await openDiscuss(channelId);
-    await contains(".o-mail-Message", { count: 25 });
-    await contains(".o-mail-Thread", { scroll: "bottom" });
+    await contains(".o-mail-Thread-newMessage ~ .o-mail-Message", { text: "message 100" });
+    const thread = document.querySelector(".o-mail-Thread");
+    const message = Array.from(document.querySelectorAll(".o-mail-Message")).find(
+        ({ textContent }) => textContent === "message 100"
+    );
```

Comment:
Why not just `queryFirst(".o-mail-Message:contains(message 100)")`?

Reply:
Forgot about query first and usage of contains^^ I'll just update it